lib/HeroBonus.h

```diff
@@ -986,3 +1001,42 @@ void BonusList::insert(const int position, InputIterator first, InputIterator la
 	bonuses.insert(bonuses.begin() + position, first, last);
 	changed();
 }
+
+// observers for updating bonuses based on certain events (e.g. hero gaining level)
+
+class DLL_LINKAGE IUpdater
+{
+public:
+	virtual ~IUpdater();
+
+	virtual bool update(Bonus & b, const CBonusSystemNode & context) const = 0;
+	virtual std::string toString() const;
+	virtual JsonNode toJsonNode() const = 0;
+
+	template <typename Handler> void serialize(Handler &h, const int version)
+	{
+	}
+};
+
+struct DLL_LINKAGE ScalingUpdater : public IUpdater

```

Comment:
As long as `IUpdater` declared with `class` descendants should be consistent.

Reply:
fixed.